hom/utils.py

```diff
@@ -120,37 +68,37 @@ async def create_ticket_for_user(
     example_url: t.Optional[str] = None,
 ) -> discord.TextChannel:
     assert interaction.guild
-    assert isinstance(interaction.user, discord.Member)
-    existing_ticket_channel = get_user_ticket_channel(interaction.guild, interaction.user)
 
+    existing_ticket_channel = get_user_ticket_channel(interaction.guild, interaction.user)
     if existing_ticket_channel:
         msg_content = f":envelope:  Click [here]({existing_ticket_channel.jump_url}) to view your open ticket."
         await interaction.followup.send(content=msg_content, ephemeral=True)
         return existing_ticket_channel
 
     channel_name = f"help-{interaction.user.display_name[:15]}"
-    tickets_category = get_category(interaction.guild, "Tickets")  # TODO: Make this channel_id?
-    mod_role = get_role_by_name(interaction.guild, "Moderator")
-    helpful_old_man_role = get_role_by_name(interaction.guild, "Helpful Old Man")
-    assert helpful_old_man_role
-    assert mod_role
+    tickets_category = get_category(interaction.guild, Config.TICKET_CATEGORY)
+    if not (mod_role := get_role(interaction.guild, Config.MOD_ROLE)):
+        await interaction.followup.send("The moderator role is missing from the server.")
+        raise RuntimeError(f"Couldn't find mod role with ID: {Config.MOD_ROLE}")
 
     new_text_channel = await interaction.guild.create_text_channel(
         name=channel_name,
         category=tickets_category,
         reason=f"{interaction.user.display_name} ({interaction.user.id}) has opened a ticket.",
         topic=button_label or "Unknown (This is a bug)",
         overwrites={
+            mod_role: discord.PermissionOverwrite(read_messages=True),
+            interaction.guild.me: discord.PermissionOverwrite(read_messages=True),
```

Comment:
Not sure if the change here is just going from interaction.user to interaction.guild.me - I'm sure both work but would be curious if there is a specific reason it was changed.

Reply:
No, originally we we're fetching the bots role by name to give permissions to the channel to that role. It was a bit clunky, instead we just give them bot itself access to the channel based on it's user. 

Additionally, you can't pass a `User` to overwrites, only a `Member` or `Role`, which is why we gets the bots `Member` (interaction.guild.me).